packages/ComponentLibrary/src/components/PrimaryTab/index.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useState } from 'react';
+import {
+  Tabs,
+  Tab,
+  Box,
+  Menu,
+  MenuItem,
+  ListItemIcon,
+  Tooltip,
+} from '@mui/material';
+import CheckIcon from '@mui/icons-material/Check';
+import { PrimaryTabsProps, TabItem } from './types';
+import { menuStyle, styles, sx, tabIndicatorProps } from './styles';
+import IconButton from '../IconButton';
+
+const PrimaryTabs: React.FC<PrimaryTabsProps> = ({ tabs, onChange, icon }) => {
+  const [selectedTab, setSelectedTab] = useState(tabs[0]?.id || '');
+  const [hoveredTab, setHoveredTab] = useState<string | null>(null);
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+
+  const handleChange = (_: React.SyntheticEvent, newValue: string) => {
+    setSelectedTab(newValue);
+    if (onChange) {
+      onChange(newValue);
+    }
+  };
+
+  const handleMenuOpen = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleMenuClose = () => {
+    setAnchorEl(null);
+  };
+
+  const handleMenuItemClick = (id: string) => {
+    setSelectedTab(id);
+    if (onChange) {
+      onChange(id);
+    }
+  };
+
+  return (
+    <Box style={styles.containerBox}>
+      <Box style={styles.tabsContainer}>
+        <Tabs
+          value={selectedTab}
+          onChange={handleChange}
+          variant="scrollable"
+          scrollButtons="auto"
+          TabIndicatorProps={tabIndicatorProps}
+          aria-label="primary tabs"
+          sx={sx.tabs}>
+          {tabs.map((tab: TabItem) => {
+            const isSelected = selectedTab === tab.id;
+            const isHovered = hoveredTab === tab.id;
+
+            return (
+              <Tab
+                key={tab.id}
+                value={tab.id}
+                icon={
+                  tab.showInTab !== 'label' && tab.icon
+                    ? React.cloneElement(tab.icon as React.ReactElement, {
+                        style: {
+                          fill: isSelected
+                            ? tab.fill
+                            : isHovered
+                              ? tab.hoverFill
+                              : tab.fill,
+                          transition: 'fill 0.3s',
+                        },
+                      })
+                    : undefined
+                }
+                label={tab.showInTab !== 'icon' ? tab.label : undefined}
+                iconPosition="start"
+                onMouseEnter={() => setHoveredTab(tab.id)}
+                onMouseLeave={() => setHoveredTab(null)}
```

Comment:
Do not use arrow functions on props/JSX section!

Why you shouldn't use inline arrow functions in JSX props

Using arrow functions or binding in JSX is a bad practice that hurts performance, because the function is recreated on each render.

https://stackoverflow.com/a/36677798